user-service/src/main/java/uk/gov/gchq/palisade/service/user/stream/config/AkkaRunnableGraph.java

```diff
@@ -0,0 +1,111 @@
+import akka.kafka.javadsl.Consumer.Control;
+import akka.kafka.javadsl.Consumer.DrainingControl;
+import akka.stream.ActorAttributes;
+import akka.stream.Materializer;
+import akka.stream.Supervision;
+import akka.stream.Supervision.Directive;
+import akka.stream.javadsl.RunnableGraph;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import scala.Function1;
+
+import uk.gov.gchq.palisade.service.user.model.UserRequest;
+import uk.gov.gchq.palisade.service.user.model.UserResponse;
+import uk.gov.gchq.palisade.service.user.service.KafkaProducerService;
+import uk.gov.gchq.palisade.service.user.service.UserServiceAsyncProxy;
+import uk.gov.gchq.palisade.service.user.stream.ConsumerTopicConfiguration;
+import uk.gov.gchq.palisade.service.user.stream.ProducerTopicConfiguration;
+import uk.gov.gchq.palisade.service.user.stream.ProducerTopicConfiguration.Topic;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Configuration for the Akka Runnable Graph used by the {@link uk.gov.gchq.palisade.service.user.UserApplication}
+ * Configures the connection between Kafka, Akka and the service
+ */
+@Configuration
+public class AkkaRunnableGraph {
+    private static final int PARALLELISM = 1;
+
+    @Bean
+    KafkaProducerService kafkaProducerService(final Sink<ProducerRecord<String, UserRequest>, CompletionStage<Done>> sink,
+                                              final ConsumerTopicConfiguration upstreamConfig,
+                                              final Materializer materializer) {
+        return new KafkaProducerService(sink, upstreamConfig, materializer);
+    }
+
+    @Bean
+    Function1<Throwable, Directive> supervisor() {
+        return ex -> Supervision.resumingDecider().apply(ex);
+    }
+
+    @Bean
+    RunnableGraph<DrainingControl<Done>> runner(
+            final Source<CommittableMessage<String, UserRequest>, Control> source,
+            final Sink<Envelope<String, UserResponse, Committable>, CompletionStage<Done>> sink,
+            final Function1<Throwable, Directive> supervisionStrategy,
+            final ProducerTopicConfiguration topicConfiguration,
+            @Qualifier("asyncUserServiceProxy") final UserServiceAsyncProxy service) {
+        // Get output topic from config
+        Topic outputTopic = topicConfiguration.getTopics().get("output-topic");
+
+        // Read messages from the stream source
+        return source
+                // Get the user from the userId, keeping track of original message and token
+                .mapAsync(PARALLELISM, (CommittableMessage<String, UserRequest> message) -> {
```

Comment:
SonarQube throws code smell here

Reply:
The smell is around the number of imports in this file rather than this `return`.